features/vaadin-surveillance-views/src/main/java/org/opennms/features/vaadin/surveillanceviews/service/DefaultSurveillanceViewService.java

```diff
@@ -0,0 +1,705 @@
+
+        stringBuffer.append("and {alias}.nodeId in (select distinct cn.nodeId from category_node cn join categories c on cn.categoryId = c.categoryId where c.categoryName in (");
+
+        for (OnmsCategory onmsCategory : colCategories) {
+
+            if (first) {
+                stringBuffer.append("'");
+                first = false;
+            } else {
+                stringBuffer.append(",'");
+            }
+
+            stringBuffer.append(onmsCategory.getName());
+            stringBuffer.append("'");
+        }
+
+        stringBuffer.append("))");
+
+        return stringBuffer.toString();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<OnmsNode> getNodesForCategories(final Set<OnmsCategory> rowCategories, final Set<OnmsCategory> colCategories) {
+        return m_transactionOperations.execute(new TransactionCallback<List<OnmsNode>>() {
+            @Override
+            public List<OnmsNode> doInTransaction(TransactionStatus transactionStatus) {
+                CriteriaBuilder criteriaBuilder = new CriteriaBuilder(OnmsNode.class);
+                criteriaBuilder.sql(createQuery(rowCategories, colCategories));
+                return m_nodeDao.findMatching(criteriaBuilder.toCriteria());
+            }
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<OnmsAlarm> getAlarmsForCategories(final Set<OnmsCategory> rowCategories, final Set<OnmsCategory> colCategories) {
+        return m_transactionOperations.execute(new TransactionCallback<List<OnmsAlarm>>() {
+            @Override
+            public List<OnmsAlarm> doInTransaction(TransactionStatus transactionStatus) {
+                final CriteriaBuilder criteriaBuilder = new CriteriaBuilder(OnmsAlarm.class);
+
+                criteriaBuilder.alias("node", "node");
+                criteriaBuilder.ne("node.type", "D");
+
+                criteriaBuilder.limit(100);
+                criteriaBuilder.distinct();
+
+                criteriaBuilder.sql(createQuery(rowCategories, colCategories));
+
+                return m_alarmDao.findMatching(criteriaBuilder.toCriteria());
+            }
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<OnmsNotification> getNotificationsForCategories(final Set<OnmsCategory> rowCategories, final Set<OnmsCategory> colCategories, final Map<OnmsNotification, String> customSeverity) {
+        return m_transactionOperations.execute(new TransactionCallback<List<OnmsNotification>>() {
+            @Override
+            public List<OnmsNotification> doInTransaction(TransactionStatus transactionStatus) {
+                Date fifteenMinutesAgo = new Date(System.currentTimeMillis() - (15 * 60 * 1000));
+                Date oneWeekAgo = new Date(System.currentTimeMillis() - (7 * 24 * 60 * 60 * 1000));
+
+                customSeverity.clear();
+
+                List<OnmsNotification> notifications = new ArrayList<>();
```

Comment:
I used the same code from the existing dashboard in class org.opennms.dashboard.server.DefaultSurveillanceService.